site/src/github.rs

```diff
@@ -32,42 +33,129 @@ pub async fn get_authorized_users() -> Result<Vec<usize>, String> {
+    for rollup_merge in rollup_merges {
+        // Fetch the rollup merge commit which should have two parents.
+        // The first parent is in the chain of rollup merge commits all the way back to `previous_master`.
+        // The second parent is the head of the PR that was rolled up. We want the second parent.
+        let commit = client.get_commit(&rollup_merge.sha).await.map_err(|e| {
+            format!(
+                "Error getting rollup merge commit '{}': {e:?}",
+                rollup_merge.sha
+            )
+        })?;
+        assert!(
+            commit.parents.len() == 2,
+            "What we thought was a merge commit was not a merge commit. sha: {}",
+            rollup_merge.sha
+        );
+        let rolled_up_head = &commit.parents[1].sha;
+
+        // Reset perf-tmp to the previous master
+        client
+            .update_branch("perf-tmp", previous_master)
+            .await
+            .map_err(|e| format!("Error updating perf-tmp with previous master: {e:?}"))?;
+
+        // Merge in the rolled up PR's head commit into the previous master
+        let sha = client
+            .merge_branch("perf-tmp", rolled_up_head, "merge")
+            .await
+            .map_err(|e| format!("Error merging commit into perf-tmp: {e:?}"))?;
+
+        // Force the `try-perf` branch to point to what the perf-tmp branch points to
+        client
+            .update_branch("try-perf", &sha)
+            .await
+            .map_err(|e| format!("Error updating the try-perf branch: {e:?}"))?;
+
+        mapping.push((rollup_merge, sha));
+        // Wait to ensure there's enough time for GitHub to checkout these changes before they are overwritten
+        tokio::time::sleep(std::time::Duration::from_secs(15)).await
+    }
+
+    Ok(mapping)
+}
+
+lazy_static::lazy_static! {
+    static ref ROLLUP_PR_NUMBER: regex::Regex =
+        regex::Regex::new(r#"^Auto merge of #(\d+)"#).unwrap();
+}
+
+// Gets the pr number for the associated rollup PR message. Returns None if this is not a rollup PR
+pub async fn rollup_pr(client: &client::Client, message: &str) -> Result<Option<u32>, String> {
+    if !message.starts_with("Auto merge of") {
+        return Ok(None);
+    }
+
+    let number = ROLLUP_PR_NUMBER
+        .captures(&message)
+        .and_then(|c| c.get(1))
+        .map(|m| {
+            println!("{}", m.as_str());
+            m.as_str().parse::<u64>()
+        })
+        .transpose()
+        .map_err(|e| format!("Error parsing PR number from '{message}': {e:?}"))?;
+
+    let number = match number {
+        Some(n) => n,
+        None => return Ok(None),
+    };
+
+    let issue = client
+        .get_issue(number)
+        .await
+        .map_err(|e| format!("Error fetching PR #{number} {e:?}"))?;
+
+    Ok(issue
+        .labels
+        .iter()
+        .any(|l| l.name == "rollup")
+        .then(|| issue.number))
+}
+
 // Returns the PR number
 pub async fn pr_and_try_for_rollup(
```

Comment:
Presumably we'll be deleting this entirely soon, but that can be a follow-up PR.